web_src/js/features/notification.js

```diff
@@ -27,35 +45,108 @@ export function initNotificationCount() {
 
   if (NotificationSettings.EventSourceUpdateTime > 0 && !!window.EventSource) {
     // Try to connect to the event source first
-    const source = new EventSource(`${AppSubUrl}/user/events`);
-    source.addEventListener('notification-count', async (e) => {
-      try {
-        const data = JSON.parse(e.data);
-
-        const notificationCount = $('.notification_count');
-        if (data.Count === 0) {
-          notificationCount.addClass('hidden');
-        } else {
-          notificationCount.removeClass('hidden');
-        }
 
-        notificationCount.text(`${data.Count}`);
-        await updateNotificationTable();
-      } catch (error) {
-        console.error(error);
-      }
-    });
-    source.addEventListener('logout', async (e) => {
-      if (e.data !== 'here') {
-        return;
-      }
-      source.close();
-      window.location.href = AppSubUrl;
-    });
-    window.addEventListener('beforeunload', () => {
-      source.close();
-    });
-    return;
+    if (window.SharedWorker && NotificationSettings.UseSharedWorker) {
+      // const {default: Worker} = await import(/* webpackChunkName: "eventsource" */'./eventsource.sharedworker.js');
+      // const worker = Worker('notification');
+      const worker = new SharedWorker('js/eventsource.sharedworker.js', 'notification-worker');
```

Comment:
You could use a path relative to the script if the filename ends with `.worker.js` (which makes it use `worker-loader` match it). For example:

```js
const {default: SharedWorker} = await import(/* webpackChunkName: "eventsource" */'./notification.worker.js');

Reply:
Actually, scrap that, `worker-loader` does not support loading shared workers yet but https://github.com/webpack-contrib/worker-loader/pull/247 was merged so it should be in the next release.

Reply:
That creates a Worker and not a SharedWorker

Reply:
Until then, I think we need to do this to get the correct script path (`__webpack_public_path__` is set previously in `publicpath.js`):

```js
const worker = new SharedWorker(`${__webpack_public_path__}js/eventsource.sharedworker.js`);

Reply:
Oh, I just noticed it was you who submitted that PR 😆